File: pippip-native-ios/API/NewAccountRequest.m
//
//  NewAccountRequest.m
//  pippip-native-ios
//
//  Created by Steve Brenneis on 10/10/17.
//  Copyright © 2017 seComm. All rights reserved.
//

#import "pippip_native_ios-Swift.h"
#import "NewAccountRequest.h"
#import "AccountManager.h"

@interface xNewAccountRequest ()
{
    SessionState *sessionState;
}
@end

@implementation xNewAccountRequest

- (instancetype)init {
    self = [super init];

    sessionState = [[SessionState alloc] init];
    return self;
}

- (NSDictionary*)restPacket {

    NSMutableDictionary *packet = [[NSMutableDictionary alloc] init];
    [packet setObject:[NSString stringWithFormat:@"%d", sessionState.sessionId]
               forKey:@"sessionId"];
    packet[@"userPublicKey"] = sessionState.userPublicKeyPEM;
    packet[@"publicId"] = sessionState.publicId;

    return packet;

}

- (double)restTimeout {
    return 10.0;
}

- (NSString*)restPath {

    if (AccountManager.production) {
        return @"/authenticator/new-account-request";
    }
    else {
        return @"/new-account-request";
    }
    
}

@end

File: pippip-native-ios/API/NewAccountRequest.swift
//
//  NewAccountRequest.swift
//  pippip-native-ios
//
//  Created by Steve Brenneis on 5/30/18.
//  Copyright © 2018 seComm. All rights reserved.
//

import UIKit
import ObjectMapper

class NewAccountRequest: NSObject, APIRequestProtocol {
    
    var path: String {
        if AccountSession.production {
            return "/authenticator/new-account-request"
        }
        else {
            return "/new-account-request"
        }
    }
    var timeout: Double = 10.0
    var sessionId: Int32?
    var authToken: Int64?
    var publicId: String?
    var userPublicKey: String?
    
    var sessionState = SessionState.instance

    override init() {

        sessionId = sessionState.sessionId
        publicId = sessionState.publicId
        userPublicKey = sessionState.userPublicKeyPEM

    }

    required init?(map: Map) {
        
    }

    func mapping(map: Map) {
        sessionId <- map["sessionId"]
        publicId <- map["publicId"]
        userPublicKey <- map["userPublicKey"]
    }
    
}
